From 1567338b2084e89db6009a95240976e06a42358f Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Thu, 13 Feb 2020 14:02:49 +0200
Subject: [PATCH 1/3] sys/boot: Objcopy more sections to final *.efi objects.

 Unbreaks uefi boot with when compiled with ld.gold from binutils234.
---
 sys/boot/efi/boot1/Makefile  | 1 +
 sys/boot/efi/loader/Makefile | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sys/boot/efi/boot1/Makefile b/sys/boot/efi/boot1/Makefile
index eecf48fd8c..d8883e3b77 100644
--- a/sys/boot/efi/boot1/Makefile
+++ b/sys/boot/efi/boot1/Makefile
@@ -82,6 +82,7 @@ boot1.efi: ${PROG}
 	fi
 	${OBJCOPY} -j .peheader -j .text -j .sdata -j .data \
 		-j .dynamic -j .dynsym -j .rel.dyn \
+		-j .got -j .got.plt -j .bss \
 		-j .rela.dyn -j .reloc \
 		--subsystem efi-app \
 		--output-target=${EFI_TARGET} ${.ALLSRC} ${.TARGET}
diff --git a/sys/boot/efi/loader/Makefile b/sys/boot/efi/loader/Makefile
index 33aa9524ff..d85175f579 100644
--- a/sys/boot/efi/loader/Makefile
+++ b/sys/boot/efi/loader/Makefile
@@ -90,6 +90,7 @@ loader.efi: ${PROG}
 	# No "-j .eh_frame", otherwise the loader doesn't run.
 	${OBJCOPY} -j .peheader -j .text -j .sdata -j .data \
 		-j .dynamic -j .dynsym -j .rel.dyn \
+		-j .got -j .got.plt -j .bss \
 		-j .rela.dyn -j .reloc -j set_Xcommand_set \
 		--subsystem efi-app \
 		--output-target=${EFI_TARGET} ${.ALLSRC} ${.TARGET}
-- 
2.25.0

